executor/memtable_reader.go

```diff
@@ -926,74 +926,77 @@ func (e *hotRegionsHistoryRetriver) retrieve(ctx context.Context, sctx sessionct
-		return nil, nil
-	}
-	row := make([]types.Datum, len(infoschema.TableTiDBHotRegionsHistoryCols))
-	updateTimestamp := time.Unix(hisHotRegion.UpdateTime/1000, (hisHotRegion.UpdateTime%1000)*int64(time.Millisecond))
+) ([][]types.Datum, error) {
+	regionsInfo := []*helper.RegionInfo{
+		{
+			ID:       int64(hisHotRegion.RegionID),
+			StartKey: hisHotRegion.StartKey,
+			EndKey:   hisHotRegion.EndKey,
+		}}
+	regionsTableInfos := tikvHelper.ParseRegionsTableInfos(regionsInfo, tables)
+
+	var rows [][]types.Datum
+	// Ignore row without corresponding schema.
+	if tableInfos, ok := regionsTableInfos[int64(hisHotRegion.RegionID)]; ok {
+		for _, tableInfo := range tableInfos {
+			updateTimestamp := time.Unix(hisHotRegion.UpdateTime/1000, (hisHotRegion.UpdateTime%1000)*int64(time.Millisecond))
+			if updateTimestamp.Location() != tz {
+				updateTimestamp.In(tz)
+			}
+			updateTime := types.NewTime(types.FromGoTime(updateTimestamp), mysql.TypeTimestamp, types.MinFsp)
+			row := make([]types.Datum, len(infoschema.TableTiDBHotRegionsHistoryCols))
+
+			row[0].SetMysqlTime(updateTime)
+			row[1].SetString(strings.ToUpper(tableInfo.DB.Name.O), mysql.DefaultCollationName)
+			row[2].SetString(strings.ToUpper(tableInfo.Table.Name.O), mysql.DefaultCollationName)
+			row[3].SetInt64(tableInfo.Table.ID)
+			if tableInfo.IsIndex {
+				row[4].SetString(strings.ToUpper(tableInfo.Index.Name.O), mysql.DefaultCollationName)
+				row[5].SetInt64(tableInfo.Index.ID)
+			} else {
+				row[4].SetNull()
+				row[5].SetNull()
+			}
+			row[6].SetInt64(int64(hisHotRegion.RegionID))
+			row[7].SetInt64(int64(hisHotRegion.StoreID))
+			row[8].SetInt64(int64(hisHotRegion.PeerID))
+			if hisHotRegion.IsLearner {
+				row[9].SetInt64(1)
+			} else {
+				row[9].SetInt64(0)
+			}
+			if hisHotRegion.IsLeader {
+				row[10].SetInt64(1)
+			} else {
+				row[10].SetInt64(0)
+			}
 
-	if updateTimestamp.Location() != tz {
-		updateTimestamp.In(tz)
-	}
-	updateTime := types.NewTime(types.FromGoTime(updateTimestamp), mysql.TypeTimestamp, types.MinFsp)
-	row[0].SetMysqlTime(updateTime)
-	row[1].SetString(strings.ToUpper(f.DBName), mysql.DefaultCollationName)
-	row[2].SetString(strings.ToUpper(f.TableName), mysql.DefaultCollationName)
-	row[3].SetInt64(f.TableID)
-	if f.IndexName != "" {
-		row[4].SetString(strings.ToUpper(f.IndexName), mysql.DefaultCollationName)
-		row[5].SetInt64(f.IndexID)
-	} else {
-		row[4].SetNull()
-		row[5].SetNull()
-	}
-	row[6].SetInt64(int64(hisHotRegion.RegionID))
-	row[7].SetInt64(int64(hisHotRegion.StoreID))
-	row[8].SetInt64(int64(hisHotRegion.PeerID))
-	if hisHotRegion.IsLearner {
-		row[9].SetInt64(1)
-	} else {
-		row[9].SetInt64(0)
-	}
-	if hisHotRegion.IsLeader {
-		row[10].SetInt64(1)
-	} else {
-		row[10].SetInt64(0)
+			row[11].SetString(strings.ToUpper(hisHotRegion.HotRegionType), mysql.DefaultCollationName)
+			if hisHotRegion.HotDegree != 0 {
+				row[12].SetInt64(hisHotRegion.HotDegree)
+			} else {
+				row[12].SetNull()
```

Comment:
Why set null? can we directly set to 0?

Reply:
Aggree, change to set it's origin value.